src/internal/Notification.ts

```diff
@@ -109,22 +193,20 @@ export class Notification<T> {
     throw new Error('unexpected notification kind value');
   }
 
-  private static completeNotification: Notification<any> = new Notification('C');
-  private static undefinedValueNotification: Notification<any> = new Notification('N', undefined);
-
+  private static completeNotification: Notification<never> & CompleteNotification = new Notification('C') as any;
   /**
    * A shortcut to create a Notification instance of the type `next` from a
    * given value.
    * @param {T} value The `next` value.
    * @return {Notification<T>} The "next" Notification representing the
    * argument.
    * @nocollapse
+   * @deprecated remove in v8. It is NOT recommended to create instances of `Notification` directly
+   * and use them. Rather, try to create POJOs matching the signature outlined in {@link ObservableNotification}.
+   * For example: `{ kind: 'N', value: 1 }`, `{kind: 'E', error: new Error('bad') }`, or `{ kind: 'C' }`.
    */
-  static createNext<T>(value: T): Notification<T> {
-    if (typeof value !== 'undefined') {
-      return new Notification('N', value);
-    }
-    return Notification.undefinedValueNotification;
+  static createNext<T>(value: T): Notification<T> & NextNotification<T> {
+    return new Notification('N', value) as any;
```

Comment:
Consider using an assertion instead of `any` and, maybe, an implicit return type:

```ts
  static createNext<T>(value: T) {
    return new Notification('N', value) as Notification<T> & NextNotification<T>;
}
```